### ISSUE TITLE
Server-side rendered CardSection does not add negative margins

### ISSUE THREAD

**Issue by rafbgarcia** [mantinedev/mantine#5742]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.5.1

### What package has an issue?

@mantine/core

### What framework do you use?

Next.js

### In which browsers you can reproduce the issue?

Chrome

### Describe the bug

In the example below, I'm rendering the same code.

The only difference is the annotation "use client" in the first Card.

## Screenshot

<img width="1055" alt="Window_and_Aliex_profile___GrowinCo" src="https://github.com/mantinedev/mantine/assets/1904314/2ff5973a-5145-4db1-8e77-489b3f86e8b6">

## Code

```tsx
"use client"

import { Card, CardSection, Text } from "@mantine/core"

export const CardSectionn = () => {
  return (
    <Card withBorder shadow="sm" mb="lg">
      <CardSection bg="orange" h={150}>
        "use client" - Header
      </CardSection>

      <Text>"use client" - Body</Text>

      <CardSection bg="orange">"use client" - Footer</CardSection>
    </Card>
  )
}
```

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by rilrom** [comment]
It looks like `data-first-section="true"` and `data-last-section="true"` are not being applied to `CardSection` in the DOM when used without `use client`.

**Comment by rtivital** [comment]
Fixed in [7.5.3](https://github.com/mantinedev/mantine/releases/tag/7.5.3)

**Comment by rafbgarcia** [comment]
Thanks @rtivital !! ❤️
